src/Serokell/Util/Trace.hs

```diff
@@ -21,7 +21,7 @@ class VarArgTrace fmt f | fmt -> f where
     traceIdFHelper :: fmt -> f
 
 instance {-# OVERLAPPING #-}
-         Print p => VarArgTrace (x -> p) (x -> x) where
+         VarArgTrace (x -> Text) (x -> x) where
```

Comment:
Why?

Reply:
https://github.com/serokell/universum/pull/195

Reply:
Then I suggest changing the constraint `Print p` to `p ~ Text` instead. Changing the matching logic can make the type inference worse.

Reply:
Good idea, will do.

Reply:
@gromakovsky Wanna do it?

Reply:
Yes, I have a tab with this PR open since the time when this comment was written. So sad I was constantly busy with Tezos-related stuff since then. Let me try to do it this week.